rbac/management/utils.py

```diff
@@ -104,11 +104,11 @@ def access_for_roles(roles, param_applications):
     """Gathers all access for the given roles and application(s)."""
     access = []
     for role in set(roles):
-        role_access = set(role.access.all())
-        for access_item in role_access:
-            if (not param_applications) or (access_item.permission_application() in param_applications):
-                access.append(access_item)
-    return access
+        if param_applications:
+            access += Access.objects.filter(role=role, permission__application__in=param_applications.split(","))
```

Comment:
Split `param_applications.split(",")` once in the beginning and reuse each iteration.

Reply:
This is resolved now!